tests/validation/tests/v3_api/common.py

```diff
@@ -990,30 +994,36 @@ def check_cluster_state(etcd_count):
     assert len(components) == 0
 
 
-def validate_dns_record(pod, record, expected):
+def validate_dns_record(pod, record, expected, port=TEST_IMAGE_PORT):
     # requires pod with `dig` available - TEST_IMAGE
     host = '{0}.{1}.svc.cluster.local'.format(
         record["name"], record["namespaceId"])
-    validate_dns_entry(pod, host, expected)
+    validate_dns_entry(pod, host, expected, port=port)
 
 
-def validate_dns_entry(pod, host, expected):
+def validate_dns_entry(pod, host, expected, port=TEST_IMAGE_PORT):
     if is_windows():
         validate_dns_entry_windows(pod, host, expected)
         return
 
     # requires pod with `dig` available - TEST_IMAGE
-    cmd = 'ping -c 1 -W 1 {0}'.format(host)
-    ping_output = kubectl_pod_exec(pod, cmd)
+    if HARDENED_CLUSTER:
+        cmd = 'curl -vs {}:{} 2>&1'.format(host, port)
```

Comment:
why can not we use `ping` for the hardened cluster?

Reply:
The `ping` utility requires the binary to be owned by root, and since we have to run this image specifically as unprivileged -- not root -- it means ping doesn't actually work in it

Reply:
We can do ownership changes but it still fails on the ping socket unfortunately, so we just use curl in this case

Reply:
ok, it makes sense. Thanks for the explanation.